### ISSUE TITLE
Validation error

### ISSUE THREAD

**Issue by angay9** [thedevdojo/voyager#2384]
- Laravel Version: 5.5

- Voyager Version: 1.0
- PHP Version: 7.1

### Description:
Hi. I am facing an issue with validation. Basically, I have one field for the BREAD named Flat Area. When I add a validation rule to it in the options like so:
```
{
    "validation": {
        "rule": "min:0|numeric"
    }
}
```
validation doesn't work. Only when I change the display name of the Flat Area field to flat_area, then it does work. I have looked into the code and found an issue. It's in the ```validateBread``` function inside the Voyager's base controller. It tries to validate the field from request named "Flat Area" (the value of the display_field column in the database), but since it's called "Flat Area" (the input has name="flat_area") it can't map the validation rules to this field. 
Is there a way to leave the display field as "Flat Area" but to make validator work?
Thanks.

**Comment by angay9** [comment]
I believe I found a solution to this problem.
If you change $rules[$row->display_name] to $rules[$row->field] [here](https://github.com/the-control-group/voyager/blob/1.0/src/Http/Controllers/Controller.php#L130) on  lines 130 and on line 132 it starts working fine.

**Comment by u01jmg3** [comment]
I too am facing this issue.

For info, this issue also relates to #2236 (#2205).

---

@angay9: please note your suggested fix is actually reverting code that was introduced in `v1.0.7`
- https://github.com/the-control-group/voyager/blob/v1.0.6/src/Http/Controllers/Controller.php#L130
  - `$rules[$row->field] = explode('|', $options->validation->rule);`
- https://github.com/the-control-group/voyager/blob/v1.0.7/src/Http/Controllers/Controller.php#L130
  - `$rules[$row->display_name] = explode('|', $options->validation->rule);`

**Comment by Light-Studio** [comment]
changing $rules[$row->display_name] to $rules[$row->field] is not the best solutions cause it show "validation.required" instead of the error message when you try submitting with some errors .

**The problem come from the display name** (you can edit it in the bread editor) this can't have a capital or a space in this name or the validation will not work. so for exemple **SEO Title** need to be **seo_title**

Now it should work properly but it's not very nice looking in your UI, best way i found is to manage the rewriting of the display name in the translation file. 

There is some place where the translation dosen't apply like in the additional field section in the post create/edit page and the table head in the browse pages, to make them work, replace 
**{{ $row->display_name }}** by **{{ __(('voyager.read.') . $row->display_name) }}** 

(you can change 'read' for any other section of your translation file)

at the line **64** vendor/tcg/voyager/resources/views/bread/browse.blade.php (for the browse pages)

and at the line **158** in /vendor/tcg/voyager/resources/views/posts/edit-add.blade.php (for the additional field in the create/edit page)

(btw it's always better to make a copy of these files in your ressources/views and make the change here and leave the vendor file untouched. )

Now go to your translation file and write all the change who need to be done.

**Comment by DODMax** [comment]
For me this commit is partially wrong:
https://github.com/the-control-group/voyager/commit/54a9b1289a5c839da97c2b58b3fc7d1863b674de#diff-bf550c9ef0c15a627ba85f6bca552e12

Validator is looking for `display_name` in the request instead of `field`, this is wrong.
Changes to files `resources/assets/js/app.js` and `src/Http/Controllers/Controller.php` in the commit above should be reverted.

**Comment by thill0822** [comment]
I used @DODMax fix.

This fixed the validation error when the fields are entered correctly, but when there's actually an error I'm not seeing any messages or validation. 

Any suggestions?

**Comment by emptynick** [comment]
This was fixed in #2660
